wagtail/admin/tests/test_dashboard.py

```diff
@@ -0,0 +1,404 @@
+            "Ameristralia Day",
+        ]
+        titles = [e.get_text(strip=True) for e in soup.select(".title-wrapper a")]
+        self.assertEqual(titles, expected_titles)
+
+
+class WorkflowObjectsToModerateQueryCount(WagtailTestUtils, TestCase):
+    fixtures = ["test.json"]
+
+    def setUp(self):
+        self.superuser = self.create_superuser(username="admin", password="password")
+        self.bob = self.create_user(username="bob", password="password")
+        self.moderator = self.create_user(username="moderator", password="password")
+
+        editors = Group.objects.get(name="Editors")
+        moderators = Group.objects.get(name="Moderators")
+
+        editors.user_set.add(self.bob)
+        moderators.user_set.add(self.moderator)
+
+        root = Page.get_first_root_node()
+        GroupPagePermission.objects.create(
+            group=editors, page=root, permission_type="change"
+        )
+        GroupPagePermission.objects.create(
+            group=moderators, page=root, permission_type="change"
+        )
+        GroupPagePermission.objects.create(
+            group=moderators, page=root, permission_type="publish"
+        )
+
+        editors.permissions.add(
+            Permission.objects.get(codename="change_fullfeaturedsnippet")
+        )
+        moderators.permissions.add(
+            *Permission.objects.filter(
+                codename__in=[
+                    "change_fullfeaturedsnippet",
+                    "publish_fullfeaturedsnippet",
+                ]
+            ),
+        )
+
+        workflow = Workflow.objects.first()
+        WorkflowContentType.objects.create(
+            workflow=workflow,
+            content_type=ContentType.objects.get_for_model(FullFeaturedSnippet),
+        )
+
+        # Pages workflow started by bob and locked by moderator
+        for page in Page.objects.filter(id__in=[9, 12]).specific():
+            page.save_revision()
+            workflow.start(page, self.bob)
+            # Lock it to test the lock indicator
+            page.locked = True
+            page.locked_by = self.moderator
+            page.locked_at = timezone.now()
+            page.save()
+
+        # Page workflow started by bob
+        for page in Page.objects.filter(id__in=[4, 13]).specific():
+            page.save_revision()
+            workflow.start(page, self.bob)
+
+        # Snippet workflow started by bob
+        for i in range(1, 3):
+            obj = FullFeaturedSnippet.objects.create(text=f"Some obj {i}")
+            obj.save_revision()
+            workflow.start(obj, self.bob)
+
+        self.dummy_request = get_dummy_request()
+        self.dummy_request.user = self.moderator
+
+    def test_panel_query_count(self):
+        panel = WorkflowObjectsToModeratePanel()
+        parent_context = {"request": self.dummy_request, "csrf_token": "dummy"}
+        # Warm up the cache
+        html = panel.render_html(parent_context)
+
+        with self.assertNumQueries(13):
```

Comment:
As quite evident in the number, there's still one N+1 query to be eliminated from here. It's the only one remaining out of all the panels, but this is caused by the following method:

https://github.com/wagtail/wagtail/blob/bfa29201c3a3e2167a9aa0d85c6e91ac7ccd9762/wagtail/models/__init__.py#L4022-L4052

It does some processing on the Python level, so it's quite tricky to optimise. Could probably annotate the needed objects separately and use it in the method if the annotation exists, but this is quite involved compared to the other changes so I skipped it for now.